src/anemoi/datasets/create/sources/accumulations.py

```diff
@@ -840,16 +861,45 @@ def _compute_accumulations(
 
     base_times = [t // 100 if t > 100 else t for t in base_times]
 
-    mars_date_time_steps = AccumulationClass.mars_date_time_steps(
-        dates=dates,
-        step1=step1,
-        step2=step2,
-        frequency=data_accumulation_period,
-        base_times=base_times,
-        adjust_step=adjust_step,
-        accumulations_reset_frequency=accumulations_reset_frequency,
-        user_date=user_date,
-    )
+    user_steps = request.get("step")
+    if user_steps is not None:
+        assert accumulations_reset_frequency is None
+        assert user_date is None
+        assert user_accumulation_period[0] == 0
+
+        first_date = context.actor.groups.provider.start
+        user_steps = sorted(_to_list(user_steps))
+        mars_date_time_steps = []
+
+        for valid_date in dates:
+            base, time, step = valid_date_to_base_date_given_user_step(valid_date, user_steps, first_date)
+
+            ts = AccumulationClass.mars_date_time_steps(
+                dates=[valid_date],
+                step1=step - user_accumulation_period[1],
+                step2=step,
+                frequency=data_accumulation_period,
```

Comment:
does this mean that the default accumulation period is not the length of the forecast step?